src/server/ua_subscription_events.c

```diff
@@ -0,0 +1,562 @@
+    return UA_STATUSCODE_GOOD;
+}
+
+void UA_Event_findVariableNode(UA_Server *server, UA_QualifiedName *name, size_t relativePathSize, UA_NodeId *event,
+                               UA_BrowsePathResult *out) {
+    /* get a list with all subtypes of aggregates */
+    struct getNodesHandle handle;
+    nodeList list;
+    LIST_INIT(&list);
+    handle.server = server;
+    handle.nodes = &list;
+    UA_StatusCode retval = UA_Server_forEachChildNodeCall(server, UA_NODEID_NUMERIC(0, UA_NS0ID_AGGREGATES),
+                                                          findAllSubtypesNodeIteratorCallback, &handle);
+    if (retval != UA_STATUSCODE_GOOD) {
+        out->statusCode = retval;
+    }
+
+    /* check if you can find the node with any of the subtypes of aggregates */
+    UA_Boolean nodeFound = UA_FALSE;
+    node_iteration *iter, *tmp_iter;
+    LIST_FOREACH_SAFE(iter, &list, listEntry, tmp_iter) {
+        if (!nodeFound) {
+            UA_RelativePathElement rpe;
+            UA_RelativePathElement_init(&rpe);
+            rpe.referenceTypeId = *iter->node;
+            rpe.isInverse = false;
+            rpe.includeSubtypes = false;
+            rpe.targetName = *name;
+            /* TODO: test larger browsepath perhaps put browsepath in a loop */
+            UA_BrowsePath bp;
+            UA_BrowsePath_init(&bp);
+            bp.relativePath.elementsSize = relativePathSize;
+            bp.startingNode = *event;
+            bp.relativePath.elements = &rpe;
+
+            *out = UA_Server_translateBrowsePathToNodeIds(server, &bp);
+            if (out->statusCode == UA_STATUSCODE_GOOD) {
+                nodeFound = UA_TRUE;
+            }
+        }
+        LIST_REMOVE(iter, listEntry);
+        UA_NodeId_delete(iter->node);
+        UA_free(iter);
+    }
+}
+
+UA_StatusCode UA_Server_createEvent(UA_Server *server, UA_NodeId eventType, UA_NodeId *outNodeId) {
+    UA_StatusCode retval;
+
+    /* make sure the eventType is a subtype of BaseEventType */
+    UA_NodeId hasSubtypeId = UA_NODEID_NUMERIC(0, UA_NS0ID_HASSUBTYPE);
+    UA_NodeId baseEventTypeId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+    if (!isNodeInTree(&server->config.nodestore, &eventType, &baseEventTypeId, &hasSubtypeId, 1)) {
+        UA_LOG_ERROR(server->config.logger, UA_LOGCATEGORY_USERLAND, "Event type must be a subtype of BaseEventType!");
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+    }
+
+    /* set the eventId attribute */
+    UA_ByteString eventId;
+    UA_ByteString_init(&eventId);
+    retval = UA_Event_generateEventId(server, &eventId);
+    if (retval != UA_STATUSCODE_GOOD) {
+        UA_ByteString_delete(&eventId);
+        return retval;
+    }
+
+    UA_ObjectAttributes oAttr = UA_ObjectAttributes_default;
+    oAttr.displayName.locale = UA_STRING("en-US");
+    oAttr.displayName.text = eventId;
+    oAttr.description = UA_LOCALIZEDTEXT("en-US", "An event in its most basic form.");
```

Comment:
Do we need to set the description here? Better if we can just set it to UA_STRING_NULL to save some bytes.